app/src/main/java/com/elmeradrianv/shesafe/SignUpActivity.java

```diff
@@ -0,0 +1,177 @@
+        user.put(User.LAST_NAME_KEY, lastName);
+        user.signUpInBackground(e2 -> {
+            if (e2 != null) {
+                Toast.makeText(this, "Couldn't sign up", Toast.LENGTH_SHORT).show();
+                Log.e(TAG, "signupNewUser: Signup error", e2);
+                return;
+            }
+            if (PERSONALIZED_PHOTO_PICKED) {
+                user.put(User.PROFILE_PHOTO_KEY, profilePhoto);
+                user.saveInBackground((SaveCallback) e3 -> {
+                    if (e2 != null) {
+                        Toast.makeText(this, "Couldn't sign up", Toast.LENGTH_SHORT).show();
+                        Log.e(TAG, "signupNewUser: Signup error", e2);
+                        return;
+                    }
+                });
+                PERSONALIZED_PHOTO_PICKED = false;
+                profilePhoto = null;
+            }
+            Toast.makeText(this, "Success!", Toast.LENGTH_SHORT).show();
+        });
+    }
+
+
+    private void setBtnAddProfilePhoto(Button btnAddProfilePhoto) {
+        btnAddProfilePhoto.setOnClickListener(v -> {
+            onPickPhoto();
+        });
+    }
+
+
+    // Trigger gallery selection for a photo
+    public void onPickPhoto() {
+        // Create intent for picking a photo from the gallery
+        Intent intent = new Intent(Intent.ACTION_PICK,
+                MediaStore.Images.Media.EXTERNAL_CONTENT_URI);
+
+        // If you call startActivityForResult() using an intent that no app can handle, your app will crash.
+        // So as long as the result is not null, it's safe to use the intent.
+        if (intent.resolveActivity(getPackageManager()) != null) {
+            // Bring up gallery to select a photo
+            startActivityForResult(intent, PICK_PHOTO_CODE);
+            PERSONALIZED_PHOTO_PICKED = true;
+        }
+    }
+
+    @Override
+    public void onActivityResult(int requestCode, int resultCode, Intent data) {
+        super.onActivityResult(requestCode, resultCode, data);
+        if ((data != null) && requestCode == PICK_PHOTO_CODE) {
+            Uri photoUri = data.getData();
+
+            String[] filePathColumn = {MediaStore.Images.Media.DATA};
+            Cursor cursor = this.getContentResolver().query(photoUri, filePathColumn, null, null, null);
+            cursor.moveToFirst();
+            int columnIndex = cursor.getColumnIndex(filePathColumn[0]);
+            String mCurrentPhotoPath = cursor.getString(columnIndex);
+            cursor.close();
+            File image = new File(mCurrentPhotoPath);
+            profilePhoto = new ParseFile(image);
+            // Load the selected image into a preview
+            ImageView ivProfilePhoto = findViewById(R.id.ivProfilePhoto);
+            setImage(photoUri.toString(), ivProfilePhoto);
+        }
+    }
+
+    private void setImage(String uri, ImageView imageView) {
+        int radius = 100000;
+        Glide.with(this).load(uri)
+                .apply(new RequestOptions()
```

Comment:
I think Glide uses the builder pattern so you can simplify this as:

`Glide.with(this).load(uri)
.centerCrop()
.placeHolder(R.drawable.ssplaceholder)
.transform(new RoundedCorners(radius))
.into(imageView)`

no need for the requestOptions

Reply:
I tried to change for use without RequestOptions and it marks error in the placeholder. Maybe it's for the version of Glide that I use